it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/datastream/DatastreamResourceManager.java

```diff
@@ -445,7 +456,9 @@ public synchronized Stream createStream(
                       .build())
               .build();
 
-      Stream reference = datastreamClient.createStreamAsync(request).get();
+      Stream reference =
+          Failsafe.with(retryOnCancellationException())
+              .get(() -> datastreamClient.createStreamAsync(request).get());
```

Comment:
Are these retries idempotent? This is test suite so maybe it doesn't matter, but there is a possibility of creation of orphaned streams if the retries are not idempotent.

Reply:
It will only retry if the request was cancelled, but even if there was a case where the stream creation is retried even after a successful creation, it would fail since it would have the same name. There is also a reaper script on the GCP project these tests run that removes old datastream streams, so a stream leak is very unlikely.

Reply:
Sounds good, thanks.